src/components/organisms/connections/integrations/asana/edit.tsx

```diff
@@ -19,10 +22,16 @@ export const AsanaIntegrationEditForm = () => {
 	const [lockState, setLockState] = useState(true);
 	const { hasActiveDeployments } = useCacheStore();
 	const { openModal } = useModalStore();
-	const { errors, handleSubmit, isLoading, onSubmitEdit, register, setValue } = useConnectionForm(
-		asanaIntegrationSchema,
-		"edit"
-	);
+	const { connectionVariables, control, errors, handleSubmit, isLoading, onSubmitEdit, register, setValue } =
+		useConnectionForm(asanaIntegrationSchema, "edit");
+
+	const pat = useWatch({ control, name: "pat" });
+
+	useEffect(() => {
+		if (!connectionVariables) return;
```

Comment:
Please move this line in all components into `setFormValues` function, to avoid code duplication.

Reply:
refactored